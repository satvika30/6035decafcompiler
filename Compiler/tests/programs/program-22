class Program {
	int a[10], temp[10], c[10];
	int num_trials;
	
	void shift(int rot_size) {
		int lcv;
		if (rot_size < 0) {
			rot_size = 10-((-rot_size)%10);
		}
		for lcv = 0, 10 {
			temp[(lcv+rot_size)%10] = a[lcv];
		}
		for lcv = 0, 10 {
			c[lcv] = temp[lcv];
		}
	}
	
	// print values
	void print_array() {
		int i;
		for i = 0, 10 {
			callout("printf","%d ",c[i]);
		}
		callout("printf","\n");
	}
	
	void main() {
		int i, rotation_size;
		num_trials = 10;
		// initialize array
		callout("printf","Initial array: \n");
		for i = 0, 10 {
			a[i] = callout("random");
			callout("printf","%d ",a[i]);
		}
		callout("printf","\n");
		for i = -10, num_trials {
			rotation_size = i;
			callout("printf", "rotation of %d:\n", i);
			shift(rotation_size);
			print_array();
		}
	}
}