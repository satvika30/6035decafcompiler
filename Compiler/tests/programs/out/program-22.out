Initial array: 
1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 
rotation of -10:
1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 
rotation of -9:
1189641421 1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 
rotation of -8:
596516649 1189641421 1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 
rotation of -7:
1649760492 596516649 1189641421 1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 
rotation of -6:
719885386 1649760492 596516649 1189641421 1804289383 846930886 1681692777 1714636915 1957747793 424238335 
rotation of -5:
424238335 719885386 1649760492 596516649 1189641421 1804289383 846930886 1681692777 1714636915 1957747793 
rotation of -4:
1957747793 424238335 719885386 1649760492 596516649 1189641421 1804289383 846930886 1681692777 1714636915 
rotation of -3:
1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 1804289383 846930886 1681692777 
rotation of -2:
1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 1804289383 846930886 
rotation of -1:
846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 1804289383 
rotation of 0:
1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 
rotation of 1:
1189641421 1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 
rotation of 2:
596516649 1189641421 1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 
rotation of 3:
1649760492 596516649 1189641421 1804289383 846930886 1681692777 1714636915 1957747793 424238335 719885386 
rotation of 4:
719885386 1649760492 596516649 1189641421 1804289383 846930886 1681692777 1714636915 1957747793 424238335 
rotation of 5:
424238335 719885386 1649760492 596516649 1189641421 1804289383 846930886 1681692777 1714636915 1957747793 
rotation of 6:
1957747793 424238335 719885386 1649760492 596516649 1189641421 1804289383 846930886 1681692777 1714636915 
rotation of 7:
1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 1804289383 846930886 1681692777 
rotation of 8:
1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 1804289383 846930886 
rotation of 9:
846930886 1681692777 1714636915 1957747793 424238335 719885386 1649760492 596516649 1189641421 1804289383 
